content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md

```diff
@@ -0,0 +1,67 @@
+
+For caching workloads, Valkey users will be able to increase maximum requests per second to over 1 million on multi-core machines such as AWS EC2  r7g.4xl
+
+### Benefits of High Capacity Shards
+
+Valkey's common approach for performance and memory increase is by scaling out, i.e.,
+adding more shards to the cluster. The availability of more powerful nodes adds another dimension
+of flexibility when designing applications. 
+
+First, higher-capacity shards enable the deployment of clusters
+with larger capacity and greater resilience to request surges. Additionally, for applications requiring multi-key
+transactions, more powerful shards simplify key distribution when designing the application by allowing a higher
+concentration of slots in a single shard. Finally, higher-capacity shards handling requests for a larger number
+of slots are more effective in load balancing resources during arbitrary increases in load on random slots.
+This reduces the need for costly over-provisioning.
+
+### Performance Without Compromising Simplicity
+
+Implementing multi-threading  can be a complex task. Over the years, contributors have been careful to maintain Redis’ and now Valkey’s simplicity . This ensures an API  that can continuously evolve without the need to use complex synchronization and avoid race conditions. 
+
+We designed our new multi-threading approach to delight Valkey users with intuitive simplicity. It utilizes a minimal number of synchronization mechanisms and keeps Valkey command execution single-threaded, simple, and primed for future enhancements.
+
+![io_threads high level design](/assets/media/pictures/io_threads.png)
+
+### High Level Design 
+The above diagram depicts the io_threads implementation from a high-level perspective. Io_threads are stateless worker threads that receive jobs to execute from the main thread. In Valkey 8, a job can involve reading and parsing a command from a client, writing back responses to the client, polling for IO events on TCP connections, or de-allocating memory. This leaves the main thread with more time to execute  commands. 
+
+The main thread orchestrates all the jobs spawned to the io_threads, ensuring that no race conditions occur. Io_threads can be easily added and removed by the main thread based on the current load to ensure efficient utilization of the underlying hardware. Despite the dynamic nature of io_threads, the main thread attempts to maintain thread affinity, ensuring that the same io_thread will handle IO for the same client to improve memory access locality. 
+
+Before executing commands, the main thread performs a new procedure, prefetch-commands-keys, which aims to reduce the number of external memory accesses needed when executing the commands on the main dictionary. A detailed explanation of the technique used in that procedure will be described in our next blog. 
```

Comment:
I would consider dropping this whole paragraph, since it's missing from the current set of PRs. We can produce blogs as necessary talking about our performance improvements.